Automattic/jetpack
# Story Block: Fix edit button used in block's toolbar

This PR fixes an issue with the story block toolbar edit button that was reported in  https://github.com/Automattic/jetpack/pull/19628#issuecomment-830472184
More precisely, the `render` props of the `MediaUpload` component used in the story block's toolbar is being ignored and replaced with a button having the label "Select Media".

<img width="496" alt="result" src="https://user-images.githubusercontent.com/230230/117474866-afb9b480-af5b-11eb-8448-1e37c7ada1a1.png">

#### Changes proposed in this Pull Request:
This PR attempts to fix this issue by looking at the element rendered in MediaUpload. If it's a simple button containing only text then we can replace it with our own label, otherwise we try to reuse the rendered element. In this case we check if the element's `children` props is simply a `string`. There are other approaches we could take: for instance we could whitelist `EditButton` intances. I'm open to other ideas.
Most recent work on this component: https://github.com/Automattic/jetpack/pull/19644

#### Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Edit a post containing a story block using an account connected to WPCOM (so you have external media sources available such as Google Photos or Vexel)
* Select the story block
* Notice the pencil icon is used in the block toolbar to edit the media used for the story
* Try inserting one image, then multiple media
* Make sure the "pencil" button and the "Select Media" one are still working and and display the menu with different sources on click. Make sure selecting media from any sources work as expected

<img width="315" src="https://user-images.githubusercontent.com/230230/118132996-9489f180-b400-11eb-9975-1cab0cf07db8.png"> <img width="315" alt="Screenshot 2021-05-13 at 15 24 44" src="https://user-images.githubusercontent.com/230230/118133634-5a6d1f80-b401-11eb-8024-3dc23460b1e7.png">


### Make sure there is no regressions:
* Set and unset the post featured image on a site connected to and disconnected from WordPress.com

<img width="284px" src="https://user-images.githubusercontent.com/230230/118132353-e4b48400-b3ff-11eb-8cfc-a139878ff94b.png">

* Image block: make sure the block edit button displays "Replace"

<img width="587px" src="https://user-images.githubusercontent.com/230230/118132750-55f43700-b400-11eb-9a49-baf23c565974.png">

* Gallery block: makes sure the default button for adding media displays "Select Images"

<img width="358px" src="https://user-images.githubusercontent.com/230230/118132882-78865000-b400-11eb-9c52-a0392765b4e6.png">

* Gallery block: makes sure that for each image, the button to modify it displays "Select Image"

<img width="258px" src="https://user-images.githubusercontent.com/230230/118133157-c8651700-b400-11eb-8616-1e529e63c00f.png">

* Slideshow block: the button for adding media should display "Select Images"

<img width="618px" src="https://user-images.githubusercontent.com/230230/118133303-f77b8880-b400-11eb-82d0-e800e6c54f29.png">
